src/Tests/Microsoft.Diagnostics.Monitoring.TestCommon/LoggingRunnerAdapter.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit.Abstractions;
+
+namespace Microsoft.Diagnostics.Monitoring.TestCommon
+{
+    public sealed class LoggingRunnerAdapter : IAsyncDisposable
+    {
+        private readonly CancellationTokenSource _cancellation = new();
+        private readonly ITestOutputHelper _outputHelper;
+        private readonly DotNetRunner _runner;
+        private readonly List<string> _standardErrorLines = new();
+        private readonly List<string> _standardOutputLines = new();
+
+        private bool _isDiposed;
+        private Task _standardErrorTask;
+        private Task _standardOutputTask;
+
+        public Dictionary<string, string> Environment { get; } = new();
+
+        public Action<string> StandardErrorCallback { get; set; }
```

Comment:
Nit: did you want these to be events rather than delegates?

Reply:
Sure? At the end of the all, they would basically be the same, just that this is not multicast (by default).